test/gui/shared/steps/steps.py

```diff
@@ -350,3 +350,16 @@ def step(context):
         waitForObjectExists(names.oCC_ShareLinkWidget_checkBox_password_QProgressIndicator).visible,
         False
     )
+  
+@When('user "|any|" opens the sharing dialog of "|any|" to delete the folder shared using the client-UI')
+def step(context, receiver, resource):
+    resource = substituteInLineCodes(context, resource).replace('//','/')
+    waitFor(lambda: isFolderSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    waitFor(lambda: shareResource(resource), context.userData['clientSyncTimeout'] * 1000)   
+    
+
+@Then('"|any|" should be displayed')
```

Comment:
this step is very generic and I suspect cannot be used everywhere. Can you make it a bit more descriptive?